core/src/main/java/org/arquillian/smart/testing/ObjectMerger.java

```diff
@@ -0,0 +1,127 @@
+package org.arquillian.smart.testing;
+
+import java.lang.reflect.Array;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class ObjectMerger<T> {
+
+    private T first;
+    private T second;
+    private T third;
+
+    public ObjectMerger(T first, T second, T third) {
```

Comment:
Same as for `ObjectMapper` - do not expose our implementation details if not necessary